push_notifications/models.py

```diff
@@ -10,91 +10,93 @@
 
 @python_2_unicode_compatible
 class Device(models.Model):
-	name = models.CharField(max_length=255, verbose_name=_("Name"), blank=True, null=True)
-	active = models.BooleanField(verbose_name=_("Is active"), default=True,
-		help_text=_("Inactive devices will not be sent notifications"))
-	user = models.ForeignKey(SETTINGS["USER_MODEL"], blank=True, null=True)
-	date_created = models.DateTimeField(verbose_name=_("Creation date"), auto_now_add=True, null=True)
+    name = models.CharField(max_length=255, verbose_name=_("Name"), blank=True, null=True)
+    active = models.BooleanField(verbose_name=_("Is active"), default=True,
+        help_text=_("Inactive devices will not be sent notifications"))
+    user = models.ForeignKey(SETTINGS["USER_MODEL"], blank=True, null=True)
+    date_created = models.DateTimeField(verbose_name=_("Creation date"), auto_now_add=True, null=True)
 
-	class Meta:
-		abstract = True
+    class Meta:
+        abstract = True
 
-	def __str__(self):
-		return self.name or \
-			str(self.device_id or "") or \
-			"%s for %s" % (self.__class__.__name__, self.user or "unknown user")
+    def __str__(self):
+        return self.name or \
+            str(self.device_id or "") or \
+            "%s for %s" % (self.__class__.__name__, self.user or "unknown user")
 
 
 class GCMDeviceManager(models.Manager):
-	def get_queryset(self):
-		return GCMDeviceQuerySet(self.model)
+    def get_queryset(self):
+        return GCMDeviceQuerySet(self.model)
 
 
 class GCMDeviceQuerySet(models.query.QuerySet):
-	def send_message(self, message, **kwargs):
-		if self:
-			from .gcm import gcm_send_bulk_message
+    def send_message(self, message, **kwargs):
+        if self:
+            from .gcm import gcm_send_bulk_message
 
-			data = kwargs.pop("extra", {})
-			if message is not None:
-				data["message"] = message
+            data = kwargs.pop("extra", {})
+            if message is not None:
+                data["message"] = message
 
-			reg_ids = list(self.filter(active=True).values_list('registration_id', flat=True))
-			return gcm_send_bulk_message(registration_ids=reg_ids, data=data, **kwargs)
+            reg_ids = list(self.filter(active=True).values_list('registration_id', flat=True))
+            if not reg_ids:
+                return
+            return gcm_send_bulk_message(registration_ids=reg_ids, data=data, **kwargs)
```

Comment:
Wouldn't it be easier to put the check in the gcm_send_bulk_message method?  That would remove the need to check for None every time you want to call the method.